### PR TITLE
Api end point

### PR DESCRIPTION
**Pull request by IamYunusAli**
In this pull request me and @Profsain have done the following project requirements.

- Create an API endpoint to list all posts for a user.
posts

- Create an API endpoint to list all comments for a user's post.
comments

- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should **not** be set by passing a parameter.
comments post

- Add authentication to your API endpoints.
auth

- Your API endpoints should receive JSON and respond to JSON as well.

### CODE REVIEW

File: app/controllers/api/authentication_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Api::AuthenticationController < Api::ApiController
+  skip_before_action :authenticate_token!
+
+  def create
+    user = User.find_by(email: params[:user][:email])
+    if user.valid_password? params[:user][:password]
+      render json: { token: JsonWebToken.encode(sub: user.id) }
+    else
+      render json: { errors: ['Invalid email or password'] }
+    end
```
Comment (BertrandConxy) [inline app/controllers/api/authentication_controller.rb:11]: - [ ] Good job on making all these API endpoints 👏 👏 However from the project requirements, you are expected to create an API endpoint for **user registration** --> POST: `/register` or `/signup` Therefore, I kindly suggest that you can create it too 😃 👍

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @IamYunusAli, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used
- good PR


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/authentication_controller.rb (hunk 1 above)
Comment (IamYunusAli) [reply app/controllers/api/authentication_controller.rb:11]: @BertrandConxy kindly can you check the requirements correctly I don't see anything that say the creation of user.
[Here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_api_endpoints.md#project-requirements)

**Review by nicupop729 — COMMENTED** [review]
Hello @Profsain & @IamYunusAli  👋,

_This is an additional review as a result of the second opinion request._

Kudos for the good work you have done for this milestone :clap:

Though, please note that the previous reviewer was right in making that requirement.
Here is a screenshot from the project requirements where is mentioned exactly what he said:
image

I noticed that you pushed other commits after requesting an SO, so when you feel ready, kindly ask for another review from your dashboard, so a reviewer can check if the requirement asked by the first reviewer is correctly implemented.

Good luck & Keep rocking :four_leaf_clover:  🚀!

**Comment by IamYunusAli** [comment]
@nicupop729 Adding Authentication to the API end point does not mean creating your user. everything was implemented before we asked for a review I don't know why it's showing up after the comment given by @BertrandConxy.

**Review by BenMukebo — APPROVED** [review]
### Hi @IamYunusAli , 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**